### ISSUE TITLE
[Crash] Crashing after placing two chunk loaders at the same place

### ISSUE THREAD

**Issue by double-pi** [SuperMartijn642/ChunkLoaders#137]
**Version Info**
- Minecraft, 1.21.1
- Chunk Loaders, 1.2.8

**Are you using OptiFine?:** no

**Steps to Reproduce**
Unclear, we have been using this pack for a while and it was okay.
Then, we tried putting two chunk loaders in the same chunk, each of them was owned by a different person.
We did it since it seemed like when the player is offline, its chunk loaders stop working.
We removed them, but since then, when leaving the area, the game crashes and sometimes rolls back.

**Crash report (~/logs/latest.log)**
[Crash Report](https://mclo.gs/xpCjUEh)

**Comment by SuperMartijn642** [comment]
The error in the log occurred when the server was iterating over some set of objects and the set was modified during the iteration. Typically this is caused by something modifying the set from a thread other than the main server thread. Alternatively, it could be caused by some mod adding a mixin which modifies the vanilla behaviour to update the set during iteration.

As far as I can tell, there's no indication the crash is caused by Chunk Loaders or any other specific mod for that matter. The only guess I could make is that it is caused by 'Smoothchunk mod' as that adds a mixin to the class where the error occurred.

> We did it since it seemed like when the player is offline, its chunk loaders stop working.

This is configurable in Chunk Loaders' config file. By default, it is set to deactivate a player's chunk loaders after it's been a week since they were last online. It could be that either the modpack you are using or the server operator changed that setting. It can be completely disabled, so a player's chunk loaders always stay active if you want that behaviour.

> We removed them, but since then, when leaving the area, the game crashes and sometimes rolls back.

Does this happen consistently? If so, does it not crash if you remove/disable either or both of the chunk loaders, and does happen again if you place/enable them again?

**Comment by double-pi** [comment]
Even after removing 'smooth chunk mod', configuring the chunk loaders to work regardless of player being online, and having a single chunkloader on a chunk. It still crashes when I move far away from the chunk loader, and doesn't when I remove it.

New log: https://mclo.gs/1lBv2eW

**Comment by SuperMartijn642** [comment]
Ah okay thank you!
Could you also share the `logs/latest.log` from the server? Sometimes the log contains additional information or an earlier error leading to the crash.

If I understand correctly, you have two chunk loaders in the same chunk from two different players. Does it matter whether both Chunk Loaders are there or does it also crash with just one? And is it those specific chunk loaders/that specific area or does the crash occur for different chunk loaders/areas as well?
Basically, I am trying to see what I would need to do to reproduce the crash.
Don't worry about it if it too much of a hassle to test, I will do some testing as well whenever I have the time. I am currently busy with exams, so it might be a bit.
